### DISCUSSION TITLE
What is the best way to wrap Chakara Components in a custom theme?

### DISCUSSION THREAD

**Discussion by yotam707** [chakra-ui/chakra-ui#3318] [Q&A] [answered]
Hi, I have a custom theme and I wrap each component like so:

```
import React from 'react';
import { Button as ChakraButton, ButtonProps } from '@chakra-ui/react';
import { ButtonSizes, ButtonVariants } from './consts';

type Props = Omit<ButtonProps, 'colorScheme' | 'variant' | 'size'> & {
  variant?: ButtonVariants;
  size?: ButtonSizes;
};

const Button = React.forwardRef<HTMLElement, Props>((props, ref) => (
  <ChakraButton externalRef={ref} {...props} />
));

export default Button;
```

But I'm getting some unexpected behavior in some components (like MenuButton). can you show what is the proper way to wrap a custom attraction for an exisiting component. 

Also: I tried using the custom style component with the `  const styles = useStyleConfig("Button", { size, variant })`

But it still not working properly.

**Comment by dodas** [comment, marked as answer]
Hey @yotam707!

You shouldn't (need to) wrap Chakra UI components in most cases.
In this case, it looks like you are wrapping the `Button` just to tighten the TS types to allow only your variants/sizes.
We recently added support for [generating custom theme typings](https://chakra-ui.com/docs/theming/advanced#theme-typings), that should hopefully solve your problem in a better way.

Cheers!

**Reply by Crizzooo** [reply]
@dodas I dont believe ColorSchemes are ever typed correctly though, they are typed as strings. Any color on how to provide typing for the ColorSchemes without re-wrapping all components then?

**Reply by Crizzooo** [reply]
Or more explicitly, being able to type versions of these components to only accept _our_ defined themes and sizes rather than all default Chakra color schemes.
